src/heroes/sombra/translocator.opy

```diff
@@ -1,46 +1,36 @@
 # OverPy implementation by Unmellow#1512
 
-#Global variables
-globalvar STEALTH_REFUND_FRACTION
-globalvar NEGATIVE_STATUS_PREVENTS_REFUND
-globalvar MINIMUM_TRANSLOCATOR_COOLDOWN
 
 #Player variables
 playervar stealth_timer
 playervar stealth_pvar
 #!defineMember _stealth_max_cooldown stealth_pvar[0]
-#!defineMember _ult_charge_percent_cache stealth_pvar[1]
 
-
-rule "[sombra/translocator_cooldown_reset.opy]: Sombra stealth refund variables":
-    STEALTH_REFUND_FRACTION = createWorkshopSetting(int[0:100], "Gameplay Settings", "Percent of Stealth Timer Remaining Refunded Towards Translocator Cooldown", 40, 0) / 100
-    NEGATIVE_STATUS_PREVENTS_REFUND = createWorkshopSetting(bool, "Gameplay Settings", "Negative Status Effects Cancelling Stealth Prevents Translocator Cooldown Refund", true, 1)
-    MINIMUM_TRANSLOCATOR_COOLDOWN = 4
-    
-    
+  
 rule "[sombra/translocator_cooldown_reset.opy]: When Translocator is used, begin tracking how long Sombra is in stealth":
     @Event eachPlayer
     @Condition (eventPlayer.getCurrentHero() == Hero.SOMBRA) == true
     @Condition eventPlayer.isUsingAbility2() == true
     
-    eventPlayer.stealth_timer = 5.5
+    eventPlayer.stealth_timer = OW2_SOMBRA_STEALTH_DURATION + OW2_SOMBRA_STEALTH_CAST_TIME
     waitUntil(eventPlayer.isUsingAbility2() == false, Math.INFINITY)
     chase(eventPlayer.stealth_timer, 0, rate=1, ChaseReeval.NONE)
     #Sombra can't break stealth during the cast time of stealth
-    wait(0.5, Wait.RESTART_WHEN_TRUE)
+    wait(OW2_SOMBRA_STEALTH_CAST_TIME, Wait.RESTART_WHEN_TRUE)
     #Wait until Sombra exits Stealth
-    waitUntil(any([eventPlayer.hasStatusEffect(Status.ASLEEP), eventPlayer.hasStatusEffect(Status.FROZEN), eventPlayer.hasStatusEffect(Status.HACKED), eventPlayer.hasStatusEffect(Status.KNOCKED_DOWN), eventPlayer.hasStatusEffect(Status.STUNNED), eventPlayer.hasStatusEffect(Status.ROOTED)]) or eventPlayer.isFiringPrimaryFire() or eventPlayer.isHoldingButton(Button.PRIMARY_FIRE) and not eventPlayer.isFiringSecondaryFire() or eventPlayer.getAbilityCooldown(Button.ABILITY_1) == 0 and eventPlayer.isHoldingButton(Button.ABILITY_1) or eventPlayer.isHoldingButton(Button.MELEE) or eventPlayer.isMeleeing() or eventPlayer.isUsingUltimate() or eventPlayer._ult_charge_percent_cache == 100 and eventPlayer.isHoldingButton(Button.ULTIMATE), 5)
+    waitUntil(any([eventPlayer.hasStatusEffect(Status.ASLEEP), eventPlayer.hasStatusEffect(Status.FROZEN), eventPlayer.hasStatusEffect(Status.HACKED), eventPlayer.hasStatusEffect(Status.KNOCKED_DOWN), eventPlayer.hasStatusEffect(Status.STUNNED), eventPlayer.hasStatusEffect(Status.ROOTED)]) or eventPlayer.isFiringPrimaryFire() or eventPlayer.isHoldingButton(Button.PRIMARY_FIRE) and not eventPlayer.isFiringSecondaryFire() or eventPlayer.getAbilityCooldown(Button.ABILITY_1) == 0 and eventPlayer.isHoldingButton(Button.ABILITY_1) or eventPlayer.isHoldingButton(Button.MELEE) or eventPlayer.isMeleeing() or eventPlayer.isUsingUltimate(), 5)
```

Comment:
Doesn't this cause a bug where the cooldown is not reduced when exiting stealth via emp?

Reply:
Actually no. The way that code worked is to check if you have q held from before you got enough ult charge to emp. Which since that doesn't actually cause you to use ult Just allows you to refund stealth without actually canceling the ability.

I tested after removing that line of code and everyway I could get Sombra to ult to actually cancel stealth gave me the refund. The only such example pressing q with 100% ult charge.

Reply:
tested again i still can't get an ult to happen, and cancel stealth without causing a refund marking as resolved for now.

Reply:
Tbh if you press q with ult charge full you actually activate your ult and get the refund but that has a separate check.

Reply:
Based on practice range testing if you tap q quickly with full ult charge you can exit stealth without activating emp.

Reply:
Ok so we have a choice between allowing my forbidden tech, and fixing this. The forbidden tech is niche enough that I agree we should just fix that problem. Still no reason to use a cache when we can just get the value as part of this condition.